modelmapper/loader.py

```diff
@@ -9,38 +10,121 @@ def select(fields):
         raise ImportError('Please install SQLAlchemy')
 
 
-class PostgresBulkLoaderMixin():
+class BaseLoaderMixin():
     """
-    Postgres Specific Loader that backs up raw data into S3.
-    Handles Bulk inserts and ignores when there were errors and continues
+    Base class for loaders.
+    """
+    def pre_row_insert(self, row, session, table):
+        """Override to add any logic for rows before being loaded"""
+        return row
+
+    def post_row_insert(self, row, session, table):
+        """Override to add any logic for rows after being loaded"""
+        pass
+
+    def insert_row_into_db(self, row, session, table):
+        """REQUIRED IMPLEMENTATION for how a row is loaded into the db"""
+        raise NotImplementedError("Please implement the insert_row_into_db.")
+
+    def insert_chunk_of_data_to_db(self, session, table, chunk):
```

Comment:
can just be chunk=(), so below you don't need to do the `if chunk`

Reply:
👍